cloudstack.go

```diff
@@ -410,10 +417,11 @@ func (d *Driver) Create() error {
 		return err
 	}
 	d.Id = vm.Id
-	d.PrivateIP = vm.Nic[0].Ipaddress
+	d.PrivateIP = vm.Nic[d.NetworkInterface].Ipaddress
 	if d.NetworkType == "Basic" {
 		d.PublicIP = d.PrivateIP
 	}
+	d.IPAddress = map[bool]string{true: d.PrivateIP, false: d.PublicIP}[d.UsePrivateIP]
```

Comment:
Let's try not to be too clever. In my option this would be more readable:
```suggestion
	d.IPAddress = d.PublicIP
	if d.UsePrivateIP {
		d.IPAddress = d.PrivateIP
	}
```